modules/v1/workspaces/controllers/WorkspaceController.php

```diff
@@ -33,4 +32,51 @@ public function actionGetWorkspaces()
             'query' => $query,
         ]);
     }
+
+    /**
+     * Get all users for workspace invitation
+     *
+     * @return ActiveDataProvider
+     */
+    public function actionGetAllUser()
+    {
+        $query = UserResource::find()->active();
+
+        return new ActiveDataProvider([
+            'query' => $query,
+        ]);
+    }
+
+    /**
+     *
+     */
+    public function actionInviteUsers()
+    {
+        $request = Yii::$app->request;
+        $selectedUsers = $request->post('selectedUsers');
+        $allUser = $request->post('allUser');
+        $workspaceId = $request->post('workspace_id');
+
+        $userIds = $selectedUsers ? $selectedUsers : $allUser;
+
+        if (!$userIds) {
+            return $this->validationError(Yii::t('app', 'Please select users'));
+        }
+
+        foreach ($userIds as $userId) {
```

Comment:
@skavt Lets optimize this code, thus batch insert data to User Workspace table.
First of all make one select, to get all user under this workspace (if necessary filter by $userIds) and then filter all user ids which are not it database. Afterwards prepare data and batch insert it to user workspace table. 
We something is unclear, let me know and we can have a brief call about this.